presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcStorageManager.java

```diff
@@ -318,10 +315,22 @@ public ConnectorPageSource getPageSource(
             Optional<ShardRewriter> shardRewriter = Optional.empty();
             if (transactionId.isPresent()) {
                 checkState(allColumnTypes.isPresent());
-                shardRewriter = Optional.of(createShardRewriter(fileSystem, transactionId.getAsLong(), bucketNumber, shardUuid, allColumnTypes.get()));
+                shardRewriter = Optional.of(createShardRewriter(fileSystemContext, fileSystem, orcFileTailSource, transactionId.getAsLong(), bucketNumber, shardUuid, deltaShardUuid, tableSupportsDeltaDelete, allColumnTypes.get()));
             }
-
-            return new OrcPageSource(shardRewriter, recordReader, dataSource, columnIds, columnTypes, columnIndexes.build(), shardUuid, bucketNumber, systemMemoryUsage);
+            return new OrcPageSource(
+                    this,
```

Comment:
Can you try passing the `getRowsFromUuid()` as a callback instead of passing the whole `OrcStorageManager`?